migration-to-python-3-only.md

```diff
@@ -0,0 +1,217 @@
+# Migration to Python 3 only
+
+This document discuss some of the various possibility to migrate Jupyter, and
+its reference Kernel implementation IPython, toward a Python 3 only code base.
+
+The document is not meant to debate whether the Jupyter & IPython project should migrate
+to a Python 3 only code base.
+
+# TL;DR:
+
+  - High Priority: Fix Pip to refuse to install Python 3 only packages on Python 2. Extremely
+    high priority as we want pip version to be release and propagate after
+    that.
+  - High Priority: Fix Setuptools to be able to express `requires-python` in `setup.py`.
+  - June 2016 : Release IPython 5.0 LTS
+    - Traitlets "LTS"
+    - ipykernel LTS
+    - ipython_genutils LTS
+  - January 2017 : Release IPython 6.0 Python 3 only.
+  - July 2017 : Release Notebook 6.0 Python 3 only.
+  - December 2017: IPython 7.0
+  - December 2017: End of active support IPython 5.x LTS
+  - July 2019 : Complete end of support for IPython 5.x
+
+
+
+# Statement of the problem
+
+We recognize that many users of IPython and Jupyter still need to use a Python 2
+environment, and will still need to do so for quite some time. However, as an open
+source project, there is an overhead in maintaining Python 2 and 3 compatibility,
+and this slows progress on our goals.
```

Comment:
This doesn't really explain why you want to do this. But perhaps that would be better left to a blog post from the developers.